kedro_airflow_k8s/hooks.py

```diff
@@ -0,0 +1,40 @@
+import os
+from typing import Dict, Iterable
+
+from kedro.config import ConfigLoader, TemplatedConfigLoader
+from kedro.framework.hooks import hook_impl
+
+VAR_PREFIX = "KEDRO_CONFIG_"
+
+# defaults provided so default variables ${commit_id|dirty} work for some entries
+ENV_DEFAULTS = {"commit_id": None, "branch_name": None}
+
+
+class RegisterTemplatedConfigLoaderHook:
+    """Provides config loader that can substitute $(commit_id) and $(branch_name)
+    placeholders with information taken from env variables."""
+
+    @staticmethod
+    def read_env() -> Dict:
+        config = ENV_DEFAULTS.copy()
+        overrides = dict(
+            [
+                (k.replace(VAR_PREFIX, "").lower(), v)
+                for k, v in os.environ.copy().items()
+                if k.startswith(VAR_PREFIX)
+            ]
+        )
+        config.update(**overrides)
```

Comment:
I wonder if it's possible to add a variable here, that we could use in templates to refer to` {package_name}` or `{project name} `?

Reply:
Variables that are added here are not visible inside DAG template. These have to be injected into templated config files first and then referenced from the DAG.

Reply:
I was thinking about config templates not DAG templates. What I would want to achieve is to be able to specify in config file:

`experiment_name: {package_name}`

on the other hand package name doesn't change often so it doesn't require templating.

Reply:
Yes, it possible to use it this way, given:
`export KEDRO_CONFIG_PACKAGE_NAME=some_package_name`

then in config template:
`experiment_name: ${package_name}`

As you said, for package name probably it's not that required, but this mechanism can be used for any other environment variable with `KEDRO_CONFIG_` prefix.